pkg/csi/service/node.go

```diff
@@ -774,6 +777,26 @@ func (s *service) NodeExpandVolume(
 		Interface: realMounter,
 		Exec:      realExec,
 	}
+
+	if containerOrchestratorUtility.IsFSSEnabled(ctx, common.OnlineVolumeExtend) {
+		// Fetch the current block size
+		gotBlockSizeBytes, err := getBlockSizeBytes(mounter, dev.RealDev)
+		if err != nil {
+			return nil, status.Error(codes.Internal, fmt.Sprintf("error when getting size of block volume at path %s: %v", dev.RealDev, err))
+		}
+		// Check if a rescan is required
+		if gotBlockSizeBytes < reqVolSizeBytes {
+			// If a device is expanded while it is attached to a VM, we need to rescan
+			// the device on the guest OS in order to see the modified size on the Guest OS
+			// Refer to https://kb.vmware.com/s/article/1006371
+			err = rescanDevice(ctx, dev)
+			if err != nil {
+				return nil, status.Error(codes.Internal, err.Error())
+			}
+		}
+	}
+
+	// Resize volume
```

Comment:
How about this comment? 

// Resize file system.

Reply:
Changed